spv/verifier.go

```diff
@@ -115,7 +101,7 @@ func VerifySPV() VerifyOpt {
 // The implementation of bc.BlockHeaderChain which is supplied will depend on the client
 // you are using, some may return a HeaderJSON response others may return the blockhash.
 type PaymentVerifier interface {
-	VerifyPayment(context.Context, *Envelope, ...VerifyOpt) (*bt.Tx, error)
+	VerifyPaymentWithAncestry(ctx context.Context, pTx []byte, ancestors []byte, opts ...VerifyOpt) (*bt.Tx, error)
```

Comment:
Was there a decision to make this name this verbose? If there's only one Verify func, then we may as well leave it as `VerifyPayment` for a smaller call and usable footprint